### PR TITLE
cmake: modules: Remove duplicates only from existing ZEPHYR_MODULE_NAMES

### PR DESCRIPTION
**Pull request by wopu-ot**
Remove duplicates from ZEPHYR_MODULE_NAMES only if that list
exists. That avoids cmake error "list sub-command REMOVE_DUPLICATES
requires list to be present" if ZEPHYR_MODULE_NAMES is empty.

Signed-off-by: Wolfgang Puffitsch <wopu@demant.com>

### CODE REVIEW

File: cmake/zephyr_module.cmake [hunk 1]
```diff
@@ -123,4 +123,6 @@ else()
 
 endif()
 
-list(REMOVE_DUPLICATES ZEPHYR_MODULE_NAMES)
+if(ZEPHYR_MODULE_NAMES)
```
Comment (tejlmand) [inline cmake/zephyr_module.cmake:126]:
```suggestion
if(DEFINED ZEPHYR_MODULE_NAMES) 
```

**Review by tejlmand — CHANGES_REQUESTED** [review]
Please check with `if(DEFINED ..)` instead.

File: cmake/zephyr_module.cmake (hunk 1 above)
Comment (wopu-ot) [reply cmake/zephyr_module.cmake]: Done.

**Review by tejlmand — APPROVED** [review]

**Comment by wopu-ot** [comment]
@nashif Can you maybe have a look at this?

**Comment by wopu-ot** [comment]
@tejlmand @nashif This rather trivial change is now already sitting there for two weeks. What can I do to get it merged?

**Review by nashif — APPROVED** [review]

**Pull request merged**